src/Components/NotificationsProvider.js

```diff
@@ -2,67 +2,133 @@ import { useEffect, useState } from "react";
 import NotificationsContext from "./NotificationsContext";
 import {
   collection,
-  doc,
-  getCountFromServer,
   getDocs,
   limit,
   onSnapshot,
   orderBy,
   query,
+  startAfter,
   where,
 } from "firebase/firestore";
 import { auth, db } from "./Firebase";
 import { useAuthState } from "react-firebase-hooks/auth";
 
 export default function NotificationsProvider(props) {
   const [user, loading, error] = useAuthState(auth);
-  const [notifications, setNotifications] = useState();
+  const [notifications, setNotifications] = useState([]);
   const [showMore, setShowMore] = useState(false);
   const [hideBadge, setHideBadge] = useState(true);
-  const [moreNotiRequests, setMoreNotiRequests] = useState(0);
+  const [listeningDocs, setListeningDocs] = useState([]);
+  const [lastVisible, setLastVisible] = useState();
+  const [lastVisibleListener, setLastVisibleListener] = useState();
+  const [showNewNotiButton, setShowNewNotiButton] = useState(false);
 
-  // Read initial state from Firestore, or else use blank array.
-  // It calls Firestore one time, after user has been authed.
+  // Creates realtime listener on most recent notifications
+  // Used as trigger for query of unseen data
   useEffect(() => {
     if (!user) return;
-    // Get notifications
+    // Set listener for changes
     const notisRef = collection(
       db,
       "notifications",
       user?.uid,
       "usersNotifications"
     );
-    const q = query(
-      notisRef,
-      orderBy("time", "desc"),
-      limit(5 + moreNotiRequests * 5)
-    );
-    const unsub = onSnapshot(q, (querySnapshot) => {
-      const notis = [];
-      querySnapshot.forEach((doc) => {
-        notis.push({ ...doc.data(), firestoreDocId: doc.id });
+    const q = query(notisRef, orderBy("time", "desc"), limit(5));
+    const unsub = onSnapshot(q, (documentSnapshots) => {
+      const latestNoti = [];
+      documentSnapshots.forEach((doc) => {
+        latestNoti.push({ ...doc.data(), firestoreDocId: doc.id });
       });
-      // Check if there are more notifications than being sent back.
-      if (notis.length === 5 + moreNotiRequests * 5) {
-        notis.splice(notis.length - 1);
-        setNotifications(notis);
+      setListeningDocs(latestNoti);
+      if (latestNoti.length === 5)
+        setLastVisibleListener(
+          documentSnapshots.docs[documentSnapshots.docs.length - 2]
+        );
+      else setLastVisibleListener(null);
+    });
+    return unsub;
+  }, [user]);
+
+  async function GetNotis() {
```

Comment:
style-wise, is there a specific reason this function starts with a capital letter?  that is usually reserved for classes, or react components/contexts/etc, whereas standard functions usually get `lowerCamelCase`

Reply:
Nope, just overlooked it - great catch!!